### PR TITLE
fix(core/inlines): support [=deep/for/links=]

### LINKED ISSUE #3595: xref not able to link [=ReadableStream/set up/pullAlgorithm=]
## Important info

- URL to affected spec: https://github.com/WICG/serial/pull/137/files

## Description of problem

**What happened (e.g., it crashed)?:**
The following is not linking via xref: 

 [=ReadableStream/set up/pullAlgorithm=]

**Expected behavior (e.g., it shouldn't crash):**

For it to link. 

**Optional, steps to reproduce:**

1.  Add `[=ReadableStream/set up/pullAlgorithm=]` to a document. See linking error.

### PR DESCRIPTION
**Pull request by marcoscaceres**
fixes #3595

### CODE REVIEW

File: src/core/inlines.js [hunk 1]
```diff
@@ -325,15 +325,29 @@ export function run(conf) {
...
  * https://github.com/w3c/respec/issues/2869
  */
-function splitBySlash(str, limit = Infinity) {
-  return str
-    .replace("\\/", "%%")
-    .split("/", limit)
-    .map(s => s && s.trim().replace("%%", "/"));
+function splitByFor(str) {
+  const cleanUp = str => str.replace("%%", "/").split("/").map(norm).join("/");
+  const safeString = norm(str).replace("\\/", "%%");
```
Comment (marcoscaceres) [inline src/core/inlines.js:345]: probably don't need norm() here...

File: src/core/inlines.js [hunk 2]
```diff
@@ -325,15 +325,29 @@ export function run(conf) {
...
-    .map(s => s && s.trim().replace("%%", "/"));
+function splitByFor(str) {
+  const cleanUp = str => str.replace("%%", "/").split("/").map(norm).join("/");
+  const safeString = str.replace("\\/", "%%");
+  const lastSlash = safeString.lastIndexOf("/");
+  if (lastSlash === -1) {
+    return [cleanUp(safeString)];
+  }
+  const forPart = safeString.substring(0, lastSlash);
+  const linkingText = safeString.substring(lastSlash + 1, safeString.length);
```
Comment (sidvishnoi) [inline src/core/inlines.js:351]: Nit: use `lastSlashIndex` or `lastSlashPosition`?

Comment (sidvishnoi) [inline src/core/inlines.js:351]:
```suggestion
  const [forPart, linkingText] = [safeString.slice(0, lastSlash), safeString.slice(lastSlash + 1)];
```

**Review by sidvishnoi — APPROVED** [review]
LGTM. Some nits.

File: src/core/inlines.js (hunk 2 above)
Comment (sidvishnoi) [reply src/core/inlines.js:351]: I would `/s/safeString/safeStr/` as param name is str and it's shorter 😄

File: src/core/inlines.js [hunk 3]
```diff
@@ -325,15 +325,29 @@ export function run(conf) {
...
+ * E.g., " ReadableStream / set up / pullAlgorithm ".
+ * Where "ReadableStream/set up/" is for-part, and "pullAlgorithm" is
+ * the linking-text.
+ *
+ * The for part is optional, but when present can be two or three levels deep.
+ *
  * @param {string} str
  *
  * TODO: Use negative lookbehind (`str.split(/(?<!\\)\//)`) when supported.
  * https://github.com/w3c/respec/issues/2869
```
Comment (sidvishnoi) [inline src/core/inlines.js:341]:
```suggestion
```

File: src/core/inlines.js (hunk 2 above)
Comment (marcoscaceres) [reply src/core/inlines.js:351]: lastSlashIndex seems nice

**Review by sidvishnoi — APPROVED** [review]
Ah. All good. I though we split `str`, but it was `safeStr`.

**Pull request merged**